### PR TITLE
8344891: Remove uses of sun.misc.ReflectUtil in java.desktop

### PR DESCRIPTION
**Pull request by prrace**
Remove uses of sun.misc.ReflectUtils which checked obsolete permissions

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344891](https://bugs.openjdk.org/browse/JDK-8344891): Remove uses of sun.misc.ReflectUtil in java.desktop (**Bug** - P4)

### Reviewers
 * [Alexander Zuev](https://openjdk.org/census#kizune) (@azuev-java - **Reviewer**)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22335/head:pull/22335` \
`$ git checkout pull/22335`

Update a local copy of the PR: \
`$ git checkout pull/22335` \
`$ git pull https://git.openjdk.org/jdk.git pull/22335/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22335`

View PR using the GUI difftool: \
`$ git pr show -t 22335`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22335.diff">https://git.openjdk.org/jdk/pull/22335.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22335#issuecomment-2494982731)
</details>

### CODE REVIEW

File: src/java.desktop/share/classes/com/sun/beans/finder/MethodFinder.java [hunk 1]
```diff
@@ -79,7 +78,7 @@ public static Method findMethod(Class<?> type, String name, Class<?>...args) thr
 
         try {
             Method method = CACHE.get(signature);
-            return (method == null) || isPackageAccessible(method.getDeclaringClass()) ? method : CACHE.create(signature);
+            return (method == null) ? method : CACHE.create(signature);
```
Comment (prrace) [inline src/java.desktop/share/classes/com/sun/beans/finder/MethodFinder.java:81]: This code looks wrong, meaning it was wrong already.
It is out of scope here to evaluate and fix that, so I submitted https://bugs.openjdk.org/browse/JDK-8344892

File: src/java.desktop/share/classes/java/beans/DefaultPersistenceDelegate.java [hunk 2]
```diff
@@ -27,7 +27,7 @@
 import java.util.*;
 import java.lang.reflect.*;
 import java.util.Objects;
-import sun.reflect.misc.*;
+import sun.reflect.misc.MethodUtil;
```
Comment (prrace) [inline src/java.desktop/share/classes/java/beans/DefaultPersistenceDelegate.java:30]: Now we only need this one class.

**Review by azuev-java — APPROVED** [review]
Looks good.

File: src/java.desktop/share/classes/javax/swing/UIDefaults.java [hunk 3]
```diff
@@ -1142,7 +1140,6 @@ public Object createValue(final UIDefaults table) {
                         cl = ClassLoader.getSystemClassLoader();
                     }
                 }
-                ReflectUtil.checkPackageAccess(className);
                 c = Class.forName(className, true, (ClassLoader)cl);
                 SwingUtilities2.checkAccess(c.getModifiers());
```
Comment (azvegint) [inline src/java.desktop/share/classes/javax/swing/UIDefaults.java:1144]: It looks like that it also is no longer needed(+line 1152). Maybe in a separate PR.

**Review by azvegint — COMMENTED** [review]
> forName now just calls Class.forName() and since java.desktop uses the same classloader as java.base this is equivalent so can be collapsed to Class.forName directly

I think the same is applicable for the `MethodUtil.getMethod()`, it is now just a wrapper for the `Class.getMethod()`.

```
./src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboBoxEditor.java:109:                    Method method = MethodUtil.getMethod(cls, "valueOf", new Class<?>[]{String.class});
./src/java.desktop/share/classes/javax/swing/plaf/nimbus/AbstractRegionPainter.java:456:                    Method method = MethodUtil.getMethod(c.getClass(), s, null);
```

File: src/java.desktop/share/classes/javax/swing/UIDefaults.java (hunk 3 above)
Comment (prrace) [reply src/java.desktop/share/classes/javax/swing/UIDefaults.java:1144]: Yes, this PR is just about ReflectUtils so this case will be handled in a separate PR

**Comment by prrace** [comment]
> > forName now just calls Class.forName() and since java.desktop uses the same classloader as java.base this is equivalent so can be collapsed to Class.forName directly
> 
> I think the same is applicable for the `MethodUtil.getMethod()`, it is now just a wrapper for the `Class.getMethod()`.
> 
> ```
> ./src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboBoxEditor.java:109:                    Method method = MethodUtil.getMethod(cls, "valueOf", new Class<?>[]{String.class});
> ./src/java.desktop/share/classes/javax/swing/plaf/nimbus/AbstractRegionPainter.java:456:                    Method method = MethodUtil.getMethod(c.getClass(), s, null);
> ```

This PR is just about ReflectUtils.

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**